packages/puppeteer-core/src/bidi/core/Request.ts

```diff
@@ -164,6 +164,21 @@ export class Request extends EventEmitter<{
     return this.#event.isBlocked;
   }
 
+  get resourceType(): string | undefined {
+    // @ts-expect-error non-standard attribute.
+    return this.#event['goog:resourceType'] ?? undefined;
+  }
+
+  get postData(): string {
+    // @ts-expect-error non-standard attribute.
+    return this.#event['goog:postData'] ?? '';
```

Comment:
I think the Puppeteer API expects undefined if not present. 
```suggestion
    return this.#event['goog:postData'] ?? undefined;
```